pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -198,10 +198,18 @@ func (state PipelineRunState) GetRunsResults() map[string][]v1beta1.CustomRunRes
 
 // GetChildReferences returns a slice of references, including version, kind, name, and pipeline task name, for all
 // TaskRuns and Runs in the state.
-func (state PipelineRunState) GetChildReferences() []v1beta1.ChildStatusReference {
+func (facts *PipelineRunFacts) GetChildReferences() []v1beta1.ChildStatusReference {
 	var childRefs []v1beta1.ChildStatusReference
 
-	for _, rpt := range state {
+	for _, rpt := range facts.State {
+		// try to replace the parameters of the reference result of whenexpression in the taskrun that has ended
+		if rpt.TaskRun != nil && rpt.isDone(facts) {
```

Comment:
I remember we have refactored the `ResolvedPipelineTask` (rpt) to only have TaskRuns and remove the TaskRun. 🤔

Reply:
sorry for the delay, i have rebased the main